applications/FSIapplication/python_scripts/partitioned_embedded_fsi_base_solver.py

```diff
@@ -0,0 +1,665 @@
+
+            self._get_convergence_accelerator().InitializeNonLinearIteration()
+
+            # Map the RELAXED_DISP from the structure FSI coupling interface to fluid FSI coupling interface
+            # This RELAXED_DISP is intended to be used to update the skin position before the level set update
+            # Note that we take advance of the fact that the coupling interfaces coincide in the embedded case
+            KratosMultiphysics.VariableUtils().CopyModelPartNodalVar(
+                KratosMultiphysics.RELAXED_DISPLACEMENT,
+                KratosMultiphysics.DISPLACEMENT,
+                self._get_fsi_coupling_interface_structure().GetInterfaceModelPart(),
+                self._get_fsi_coupling_interface_fluid().GetInterfaceModelPart(),
+                0)
+
+            # Update the fluid FSI coupling interface position
+            self._get_fsi_coupling_interface_fluid().UpdatePosition()
+
+            # Update the EMBEDDED_VELOCITY and solve the fluid problem
+            self._solve_fluid()
+
+            # Transfer the fluid load to the structure FSI coupling
+            self._transfer_fluid_load()
+
+            # Transfer fluid from the structure FSI coupling interface to father model part
+            if (self.level_set_type == "continuous"):
+                self._get_fsi_coupling_interface_structure().TransferValuesToFatherModelPart(KratosMultiphysics.POSITIVE_FACE_PRESSURE)
+            elif (self.level_set_type == "discontinuous"):
+                self._get_fsi_coupling_interface_structure().TransferValuesToFatherModelPart(KratosMultiphysics.POSITIVE_FACE_PRESSURE)
+                self._get_fsi_coupling_interface_structure().TransferValuesToFatherModelPart(KratosMultiphysics.NEGATIVE_FACE_PRESSURE)
+            else:
+                err_msg = 'Level set type is: \'' + self.level_set_type + '\'. Expected \'continuous\' or \'discontinuous\'.'
+                raise Exception(err_msg)
+
+            # Solve the structure problem
+            self._solve_structure()
+
+            # Compute the residual and perform the update
+            dis_residual_norm = self._get_fsi_coupling_interface_structure().Update()
+
+            # End the FSI non-linear iteration
+            self._get_convergence_accelerator().FinalizeNonLinearIteration()
+
+            # Check convergence
+            is_converged = self._check_FSI_convergence(dis_residual_norm)
+            if (is_converged):
+                return True
+
+        return False
+
+    def FinalizeSolutionStep(self):
+        # Finalize solution step
+        self.fluid_solver.FinalizeSolutionStep()
+        self.structure_solver.FinalizeSolutionStep()
+        self._get_convergence_accelerator().FinalizeSolutionStep()
+
+    def SetEchoLevel(self, structure_echo_level, fluid_echo_level):
+        self.fluid_solver.SetEchoLevel(self, fluid_echo_level)
+        self.structure_solver.SetEchoLevel(self, structure_echo_level)
+
+    def SetTimeStep(self, step):
```

Comment:
why is this needed?
Doesn't it mess up things?
```suggestion
    def __SetTimeStep(self, step):
```

Reply:
Actually I think it is not used in any place.

Reply:
Effectively, It wasn't used (removed).